website/content/en/preview/upgrading/v1-migration.md

```diff
@@ -0,0 +1,246 @@
+
+* **metadataOptions could break workloads**: If you have workload pods that are not using `hostNetworking`, the updated default `metadataOptions` could cause your containers to break when you apply new EC2NodeClasses on v1.
+
+* **Support for native Ubuntu AMI selection is dropped**: If you are using Ubuntu, be aware that Karpenter v1 no longer natively supports Ubuntu. To continue using Ubuntu in v1, you can update `amiSelectorTerms` in your EC2NodeClasses to identify Ubuntu as the AMI you want to use. See [reimplement amiFamily](https://github.com/aws/karpenter-provider-aws/pull/6569) for an example. Once you have done that, you can leave the amiFamily as Ubuntu and proceed to upgrading to v1. This will result in the following Transformation:
+   ```yaml
+   version: karpenter.k8s.aws/v1beta1
+   kind: EC2NodeClass
+   spec:
+     amiFamily: Ubuntu
+     amiSelectorTerms:
+     - id: ami-foo
+   ---
+   # Conversion Webhook Output
+   version: karpenter.k8s.aws/v1
+   kind: EC2NodeClass
+   metadata:
+     annotations:
+       compatibility.karpenter.k8s.aws/v1beta1-ubuntu: amiFamily,blockDeviceMappings
+   spec:
+     amiFamily: AL2
+     amiSelectorTerms:
+     - id: ami-foo
+     blockDeviceMappings:
+     - deviceName: 'dev/sda1'
+       rootVolume: true
+       ebs:
+         encrypted: true
+         volumeType: gp3
+         volumeSize: 20Gi
+   ```
+   **NOTE**: If you do not specify `amiSelectorTerms` before upgrading with an Ubuntu AMI, the conversion webhook will fail. If you do this, downgrade to Karpenter `v0.37.1 and add amiSelectorTerms before upgrading again.
+
+* **Migrating userData**: Consider migrating your userData ahead of the v1.0.0 upgrade. See the description of `amiSelectorTerms` below.
+
+### Before upgrading to `1.1.0`
+
+Apply the following changes to your NodePools and EC2NodeClasses, as appropriate, before upgrading them to `v1.1.0` (though okay to make these changes for `1.0.0`)
+
+* **v1beta1 support gone**: In `1.1.0`, v1beta1 is not supported. Migrate all Karpenter yaml files ([NodePools]({{<ref "../concepts/nodepools" >}}), [EC2NodeClasses]({{<ref "../concepts/nodeclasses" >}}), and so on) to v1.
+Also, know that all resources in the cluster also need to be on v1. It's possible (although unlikely) that some resources still may be stored as v1beta1 in ETCD if no writes had been made to them since the v1 upgrade.
+You could use a tool such as [kube-storage-version-migrator](https://github.com/kubernetes-sigs/kube-storage-version-migrator) to handle this.
+
+* **Remove kubelet objects from NodePools**: Check that NodePools no longer contain the kubelet-configuration annotation (`karpenter.sh/v1beta1-kubelet-conversion` annotation).
+Karpenter will crash if NodePool resources contain this annotation.
+
+* **Remove BootstrapMode annotation**: Karpenter will crash if NodePool resources contain the `BootstrapMode` annotation.
```

Comment:
@engedaam I don't think that we need to crash if this annotation is still present. Seems like the only one that we should really be crashing on is the kubelet-conversion one -- only because not crashing on it would mean that we are reasoning about the wrong configuration starting on v1.1